dask/dataframe/core.py

```diff
@@ -3134,6 +3165,62 @@ def memory_usage(self, index=True, deep=False):
         )
         return delayed(sum)(result.to_delayed())
 
+    @derived_from(pd.Series)
+    def agg(self, arg):
+        numeric_only_aggs = ["prod", "var", "std", "mean"]
+        other_aggs = ["any", "sum", "min", "max", "count"]  # size
+        supported_funcs = numeric_only_aggs + other_aggs
+
+        if isinstance(arg, str):
+            if arg in supported_funcs:
+                return getattr(self, arg)()
+            else:
+                raise ValueError(
+                    "unknown/unsupported aggregate function: {}".format(arg)
+                )
+        elif isinstance(arg, Iterable):
+            supported_aggs = [argfunc in supported_funcs for argfunc in arg]
+            if all(supported_aggs):
+                from .multi import concat
+
+                series = []
+                for aggfunc in arg:
+                    if aggfunc == "size":
+                        series.append(
+                            getattr(
+                                self.to_frame().rename(None, {self.name: aggfunc}),
+                                aggfunc,
+                            )
+                        )
+                    else:
+                        # dask.DataFrame.prod returns a value, even for object dtype
+                        if not (
+                            aggfunc in numeric_only_aggs and self.dtype == "object"
+                        ):
+                            series.append(
+                                getattr(
+                                    self.to_frame().rename(None, {self.name: aggfunc}),
+                                    aggfunc,
+                                )()
+                            )
+                return concat(series).repartition(npartitions=1)
+            else:
+                raise ValueError(
+                    "unknown/unsupported aggregate function(s): {}".format(
+                        ",".join(
+                            [
+                                agg
+                                for agg, supported in zip(arg, supported_aggs)
+                                if supported is False
+                            ]
+                        )
+                    )
+                )
```

Comment:
Took me a bit to understand this logic. I'd find the following validation logic cleaner:

```python
unsupported_aggs = [a for a in arg if a not in supported_aggs]

if not unsupported_aggs:
    # All valid, do stuff as you've done above
    ...
else:
    raise ValueError(
        "unknown/unsupported aggregate function(s): {}".format(unsupported_aggs)
    )
```
```